torlakian_data.py

```diff
@@ -0,0 +1,258 @@
+                       characters around the current character
+            - labels: boolean value that defines if a character at that position is uppercased or not"""
+            return self._data
+
+        @property
+        def size(self) -> int:
+            """Size of the dataset"""
+            return self._size
+
+        @property
+        def dataset(self) -> tf.data.Dataset:
+            """TF Dataset object"""
+            return tf.data.Dataset.from_tensor_slices(self._data)
+
+    def download(self):
+        """Downloads the corpus"""
+        
+        directory = "data"
+        path = os.path.join(directory, "allzip.zip")
+        if not os.path.exists(path):
+            os.makedirs(directory)
+            print("Downloading dataset {}...".format(path), file=sys.stderr)
+            urllib.request.urlretrieve(self.URL, filename=path)
+            print("Download finished.")
+
+        return path
+
+    def unzip(self, path):
+        """Unzipping of the corpus .zip archive"""
+
+        print("Unzipping...")
+        with zipfile.ZipFile(path, 'r') as zip_ref:
+            directory = os.path.dirname(path)
+            zip_ref.extractall(directory)
+
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if filename.endswith(".zip") and filename != os.path.basename(path):
+                full_path = os.path.join(directory, filename)
+                with zipfile.ZipFile(full_path, 'r') as zip_ref:
+                    zip_ref.extractall(directory)
+
+                os.remove(full_path)
+        
+        print("Unzipping finished.")
+    
+    def join_data(self):
+        """Joins all files in the corpus into one big dataset, 
+        then performs 70:20:10 train-val-test split and 
+        saves these into separate files"""
+
+        directory = 'data/TOR_C_TXT_transcripts'
+        all_data = []
+
+        for file in os.listdir(directory):
+            if not file.startswith("TIM_"):
+                continue
+
+            filepath = os.path.join(directory, os.fsdecode(file))
+            if not filepath.endswith(".txt"):
+                continue
+
+            with open(filepath, 'r') as f:
+                lines = f.readlines()
+                lines = lines[1:]
+                lines = [' '.join(l.split()[1:]) for l in lines]
+                all_data.extend(lines)
+
+        num_lines = len(all_data)
+        train = all_data[:int(num_lines * 0.7)]
+        val = all_data[int(num_lines * 0.7): int(num_lines * 0.9)]
+        test = all_data[int(num_lines * 0.9):]
+
+        with open('data/train.txt', 'w') as f:
+            f.write('\n'.join(train))
+        with open('data/val.txt', 'w') as f:
+            f.write('\n'.join(val))
+        with open('data/test.txt', 'w') as f:
+            f.write('\n'.join(test))
+
+    def __init__(self, window: int, alphabet_size: int = 0, clean: bool = False):
+        path = self.download()
```

Comment:
shouldn't these three funcstions be private? Or do you also expect them to be called outside of `__init__` ?

Reply:
Agreed, they should be private. Fixed